adapter/config/ingress/status.go

```diff
@@ -58,18 +60,18 @@ func NewStatusSyncer(mesh *proxyconfig.ProxyMeshConfig, client kubernetes.Interf
 	informer := cache.NewSharedIndexInformer(
 		&cache.ListWatch{
 			ListFunc: func(opts meta_v1.ListOptions) (runtime.Object, error) {
-				return client.ExtensionsV1beta1().Ingresses(options.AppNamespace).List(opts)
+				return client.ExtensionsV1beta1().Ingresses(options.WatchedNamespace).List(opts)
 			},
 			WatchFunc: func(opts meta_v1.ListOptions) (watch.Interface, error) {
-				return client.ExtensionsV1beta1().Ingresses(options.AppNamespace).Watch(opts)
+				return client.ExtensionsV1beta1().Ingresses(options.WatchedNamespace).Watch(opts)
 			},
 		},
 		&v1beta1.Ingress{}, options.ResyncPeriod, cache.Indexers{},
 	)
 
 	var publishService string
 	if mesh.IngressService != "" {
-		publishService = fmt.Sprintf("%v/%v", options.Namespace, mesh.IngressService)
+		publishService = fmt.Sprintf("%v/%v", os.Getenv("POD_NAMESPACE"), mesh.IngressService)
```

Comment:
It's not a good practice to read environment variables from within libraries. Can this be passed as a parameter to the constructor?

Reply:
Why is this not a good practice?

Then we are back to having two namespaces, and the whole point of this issue is moot.

Reply:
It leads to unexpected dependencies (e.g. ingress library also requires this environment variable and panics without it). Would it make more sense to require that ingress service in mesh is fully qualified?

Reply:
If reading env variable is not good for the reasons you mentioned, then we need to pass the pod namespace as a parameter, for the reasons I enumerated above, which leads to two namespaces, which makes the changes in this PR not needed.
Passing the fully qualified service for the ingress is a hack and would not apply properly to other uses.

Reply:
IMO there should a global context (global part of the config that can't change like where is the process running) and a local context that (may) change for each controller

Reply:
The global context is the mesh config, and we have already defined the `IngressService` name in there. Till now, it was not FQ since it's local to a namespace. If we do per-cluster installation, then the service name should be qualified.

Reply:
k so can we add the istio-namespace there too ? or simply in a bag of objects

I don't think you'd want to get a fully qualified istio-pilot.namespace.svc.cluster.local and then try to parse out namespace, let's just have the namespace in it's own IstioNamespace somewhere ?